### DISCUSSION TITLE
event_rule=<EventRule: interface update> in the webhook data

### DISCUSSION THREAD

**Discussion by smartjack** [netbox-community/netbox#17819] [General]
Hi, 

I've run into trouble with webhooks, that I can't really solve. It appears that the webhook is sent with the `event_rule=<EventRule: interface update>` parameter, which is then erroring out when the rqworker tries to process it: 

```
11:58:06 default: extras.webhooks.send_webhook(data={'id': 4, 'url': '/api/dcim/interfaces/4/', 'display_url': '/dcim/interface..., event_rule=<EventRule: interface update>, event_type='object_updated', model_name='interface', request_id=UUID('0f19932d-770d-40d5-8c84-8f61c3d15e8e'), snapshots={'prechange': {'created': '2021-08-05T00:00:00Z', 'device': 7, 'name': 'Vla..., timestamp='2024-10-21T11:58:06.221918+00:00', username='xxx') (16163bbd-0b79-4829-b663-a64026b1f274)
```

```
11:58:06 [Job 16163bbd-0b79-4829-b663-a64026b1f274]: exception raised while executing (extras.webhooks.send_webhook)
Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.11/site-packages/rq/worker.py", line 1430, in perform_job
    rv = job.perform()
         ^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.11/site-packages/rq/job.py", line 1280, in perform
    self._result = self._execute()
                   ^^^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.11/site-packages/rq/job.py", line 1317, in _execute
    result = self.func(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: send_webhook() got an unexpected keyword argument 'event_type'
```
I tried to upgrade to the latest version (using docker) but this still fails.

Is there anything I can do to fix this? I tried to rebuild the netbox docker image and everything else is working. I do not use any kind of plugins. 

Thank you

_Edit:_ I dug around the code a bit and realized how this is structured, but still the issue is there, the `send_webhook` function is not able to process the `event_type` argument...

**Comment by alexandrud** [comment]
For me, I forgot to update the worker image tag and there was a mismatch between netbox and the worker.
